pkg/authn/k8schain/k8schain.go

```diff
@@ -119,11 +121,18 @@ func NewNoClient() (authn.Keychain, error) {
 	return New(nil, Options{})
 }
 
-type lazyProvider credentialprovider.LazyAuthConfiguration
+type lazyProvider struct {
+	kc    *keychain
+	image string
+}
 
 // Authorization implements Authenticator.
 func (lp lazyProvider) Authorization() (*authn.AuthConfig, error) {
-	authConfig := credentialprovider.LazyProvide(credentialprovider.LazyAuthConfiguration(lp))
+	creds, found := lp.kc.keyring.Lookup(lp.image)
+	if !found || len(creds) < 1 {
```

Comment:
Is there a UT, by chance? :)

Reply:
Unfortunately the way the K8s credentialprovider package works it is incredibly hard to unit test this stuff, which is why the `ko`-based integration tests in the `test` sub-directory were originally written.